src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -124,44 +124,49 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
 
             _state.TcsQueryDataAvailable =
                 new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
-            _state.TcsQueryDataAvailable.Task.ContinueWith((previousTask) => 
+            
+            _state.TcsQueryDataAvailable.Task.ContinueWith((previousTask, state) => 
+            {
+                var self = (WinHttpResponseStream)state;
+                WinHttpRequestState requestState = self._state;
+                SafeWinHttpHandle requestHandle = self._requestHandle;
+                
+                if (previousTask.IsFaulted)
+                {
+                    requestState.TcsReadFromResponseStream.TrySetException(previousTask.Exception.InnerException);
+                }
+                else if (previousTask.IsCanceled || token.IsCancellationRequested)
```

Comment:
Hold on, don't merge this just yet; it looks like a closure for `token` is also being implicitly allocated here. Will patch in a moment.